openedx/core/lib/block_cache/tests/test_utils.py

```diff
@@ -0,0 +1,229 @@
+
+    def get_children(self):
+        """
+        Returns the children of the mock XBlock.
+        """
+        return [self.modulestore.get_item(child) for child in self.children]
+
+
+class MockModulestore(object):
+    """
+    A mock Modulestore to be used in unit tests, providing only the
+    minimum methods needed by the block cache framework.
+    """
+    def __init__(self):
+        self.get_items_call_count = 0
+        self.blocks = None
+
+    def set_blocks(self, blocks):
+        """
+        Updates the mock modulestore with a dictionary of blocks.
+
+        Arguments:
+            blocks ({block key, MockXBlock}) - A map of block_key
+            to its mock xBlock.
+        """
+        self.blocks = blocks
+
+    def get_item(self, block_key, depth=None):  # pylint: disable=unused-argument
+        """
+        Returns the mock XBlock (MockXBlock) associated with the
+        given block_key.
+        """
+        self.get_items_call_count += 1
+        return self.blocks.get(block_key)
+
+
+class MockCache(object):
+    """
+    A mock Cache object, providing only the minimum features needed
+    by the block cache framework.
+    """
+    def __init__(self):
+        # An in-memory map of cache keys to cache values.
+        self.map = {}
+
+    def set(self, key, val):
+        """
+        Associates the given key with the given value in the cache.
+        """
+        self.map[key] = val
+
+    def get(self, key, default=None):
+        """
+        Returns the value associated with the given key in the cache;
+        returns default if not found.
+        """
+        return self.map.get(key, default)
+
+    def set_many(self, map_):
+        """
+        For each dictionary entry in the given map, updates the cache
+        with that entry.
+        """
+        for key, val in map_.iteritems():
+            self.set(key, val)
+
+    def get_many(self, keys):
+        """
+        Returns a dictionary of entries for each key found in the cache.
+        """
+        return {key: self.map[key] for key in keys if key in self.map}
+
+    def delete(self, key):
+        """
+        Deletes the given key from the cache.
+        """
+        del self.map[key]
+
+
+class MockModulestoreFactory(object):
+    """
+    A factory for creating MockModulestore objects.
+    """
+    @classmethod
+    def create(cls, children_map):
+        """
+        Creates and returns a MockModulestore from the given
+        children_map.
+
+        Arguments:
+            children_map ({block_key: [block_key]}) - A dictionary
+                mapping a block key to a list of block keys of the
+                block's corresponding children.
+        """
+        modulestore = MockModulestore()
+        modulestore.set_blocks({
+            block_key: MockXBlock(block_key, children=children, modulestore=modulestore)
+            for block_key, children in enumerate(children_map)
+        })
+        return modulestore
```

Comment:
Nit: Why create a Factory instead of giving MockModulestore a more robust constructor?  Seems you'd get the same effect by allowing `MockModulestore.__init__` to take a `children_map`, and not have the cognitive load of an extra factory class.